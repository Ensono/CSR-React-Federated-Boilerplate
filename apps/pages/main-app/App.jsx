import React from 'runtime-library/lib/react';
import Button from 'runtime-library/Button';
import Tooltip from 'runtime-library/Tooltip';
import SampleMFE from 'sample-mfe/SampleMFE';
export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dialogVisible: false,
    };
    this.handleClick = this.handleClick.bind(this);
    this.handleSwitchVisible = this.handleSwitchVisible.bind(this);
  }
  handleClick(ev) {
    console.log(ev);
    this.setState({
      dialogVisible: true,
    });
  }
  handleSwitchVisible(visible) {
    this.setState({
      dialogVisible: visible,
    });
  }
  render() {
    return (
      <div>
        <div className="mfe-container" style={{ border: '1px solid hotpink' }}>
          <p>
            react„ÄÅreact-dom, Tooltip and Button js files hosted on <strong>runtime-library</strong>
          </p>
          <Button type="primary" />
          <Button type="warning" />
          <Tooltip content="hover me please" message="Hello,world!" />
        </div>
        <div className="mfe-container" style={{ border: '1px solid hotpink' }}>
          <p>
            MFE hosted on <strong>sample_mfe</strong>
          </p>
          <SampleMFE />
        </div>
      </div>
    );
  }
}
