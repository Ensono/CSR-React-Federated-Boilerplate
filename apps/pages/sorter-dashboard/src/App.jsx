import React from "runtime-library/lib/react"
import Button from "runtime-library/Button"
import Tooltip from "runtime-library/Tooltip"
import SampleMFE from "sample-mfe/SampleMFE"

export default class App extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div>
        <div className="mfe-container" style={{border: "1px solid hotpink"}}>
          <p>
            react„ÄÅreact-dom, Tooltip and Button js files hosted on <strong>runtime-library</strong>
          </p>
          <Button type="primary" />
          <Button type="warning" />
          <Tooltip content="hover me please" message="Hello,world!" />
        </div>
        <div className="mfe-container" style={{border: "1px solid hotpink"}}>
          <p>
            MFE hosted on <strong>sample-mfe</strong>
          </p>
          <SampleMFE />
        </div>
      </div>
    )
  }
}
