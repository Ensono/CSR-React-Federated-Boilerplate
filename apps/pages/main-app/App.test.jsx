import React from "react"
import {render} from "@testing-library/react"
import App from './App'

// jest.mock("sample-mfe/SampleMFE", () => jest.fn())

describe("App: ", () => {
    let props = {type: 'primary'}

    it("should match the snapshot for type = primary", () => {
        const {asFragment} = render(
                <App {...props} />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    
})
