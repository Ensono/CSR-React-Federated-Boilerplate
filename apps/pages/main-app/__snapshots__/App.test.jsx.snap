// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`area-dashboard:  App should match the snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  padding: 10px 16px;
  border: 1px solid #dcdfe6;
  display: inline-block;
  cursor: pointer;
  border-radius: 4px;
  position: relative;
}

.emotion-0:before {
  content: attr(data-content);
  max-width: 100%;
  box-sizing: border-box;
  position: absolute;
  background-color: #303133;
  color: #fff;
  font-size: 12px;
  border-radius: 4px;
  padding: 10px;
  left: 50%;
  bottom: 100%;
  -webkit-transform: translate(-50%,-10px);
  -moz-transform: translate(-50%,-10px);
  -ms-transform: translate(-50%,-10px);
  transform: translate(-50%,-10px);
  display: none;
}

.emotion-0:after {
  display: none;
  content: "";
  border: 6px solid transparent;
  border-top-color: #303133;
  position: absolute;
  left: 50%;
  bottom: 100%;
  -webkit-transform: translate(-50%,2px);
  -moz-transform: translate(-50%,2px);
  -ms-transform: translate(-50%,2px);
  transform: translate(-50%,2px);
}

.emotion-0:hover::after,
.emotion-0:hover::before {
  display: block;
}

<div>
    <div
      class="mfe-container"
      style="border: 1px solid hotpink;"
    >
      <p>
        react„ÄÅreact-dom, Tooltip and Button js files hosted on 
        <strong>
          runtime-library
        </strong>
      </p>
      <button
        style="margin-left: 10px; color: rgb(255, 255, 255); background-color: rgb(64, 158, 255); padding: 12px 20px; font-size: 14px; border-radius: 4px; outline: none; border: 1px solid #dcdfe6; cursor: pointer;"
      >
        primary Button
      </button>
      <button
        style="margin-left: 10px; color: rgb(255, 255, 255); background-color: rgb(230, 162, 60); padding: 12px 20px; font-size: 14px; border-radius: 4px; outline: none; border: 1px solid #dcdfe6; cursor: pointer;"
      >
        warning Button
      </button>
      <div
        class="emotion-0"
        data-content="Hello,world!"
      >
        hover me please
      </div>
    </div>
    <div
      class="mfe-container"
      style="border: 1px solid hotpink;"
    >
      <p>
        MFE hosted on 
        <strong>
          sample-mfe
        </strong>
      </p>
      <div>
        SampleMFE
      </div>
    </div>
  </div>
</DocumentFragment>
`;
